lib/promscrape/client.go

```diff
@@ -160,6 +162,18 @@ func newClient(ctx context.Context, sw *ScrapeWork) *client {
 		// See https://github.com/VictoriaMetrics/VictoriaMetrics/issues/1017#issuecomment-767235047
 		Timeout: 30 * sw.ScrapeTimeout,
 	}
+	if sw.EnableHTTP2 {
+		// golang HTTP/2 support had many problematic corner cases where
+		// dead connections would be kept and used in connection pools.
+		// https://github.com/golang/go/issues/32388
+		// https://github.com/golang/go/issues/39337
+		// https://github.com/golang/go/issues/39750
+		_, err := http2.ConfigureTransports(sc.Transport.(*http.Transport))
+		if err != nil {
+			logger.Fatalf("failed to configure net/http HTTP/1 Transport to use HTTP/2.: %s", err)
```

Comment:
This function could be called not only on init of the application, but also during runtime. Hence, this could panic during runtime. I think, we should warn that switching to http2 failed and continue using http1 transport.


```suggestion
			logger.Errorf("failed to configure HTTP/2 transport: %s", err)
```

Reply:
Right, fixed!